code/MathSpamProtectorField.php

```diff
@@ -122,24 +131,27 @@ public static function get_math_question()
     }
 
     /**
-     * Checks the given answer if it matches the addition of the saved session 
+     * Checks the given answer if it matches the addition of the saved session
      * variables.
      *
      * Users can answer using words or digits.
      *
      * @return bool
      */
-    public static function correct_answer($answer)
+    public function isCorrectAnswer($answer)
     {
-        $v1 = Session::get("mathQuestionV1");
-        $v2 = Session::get("mathQuestionV2");
 
-        Session::clear('mathQuestionV1');
-        Session::clear('mathQuestionV2');
+        $session = Controller::curr()->getRequest()->getSession();
```

Comment:
I think you can use `$this->getRequest()->getSession()` since `$this` is an instance of RequestHandler

Reply:
This fails with lack of session object if I use `$this->getRequest()->getSession()`